app/assets/javascripts/dashboard/advanced_dataset/advanced_dataset_view.js

```diff
@@ -19,27 +21,85 @@ type Props = {
   searchQuery: string,
 };
 
-class AdvancedDatasetView extends React.PureComponent<Props> {
+type State = {
+  prevSearchQuery: string,
+  sortedInfo: Object,
+};
+
+class AdvancedDatasetView extends React.PureComponent<Props, State> {
+  static getDerivedStateFromProps(nextProps: Props, prevState: State): $Shape<State> {
+    const maybeSortedInfo =
+      // Clear the sorting exactly when the search box is initially filled
+      // (searchQuery changes from empty string to non-empty string)
+      nextProps.searchQuery !== "" && prevState.prevSearchQuery === ""
+        ? {
+            sortedInfo: { columnKey: null, order: "ascend" },
+          }
+        : {};
+
+    return {
+      prevSearchQuery: nextProps.searchQuery,
+      ...maybeSortedInfo,
+    };
+  }
+
+  constructor() {
+    super();
+    this.state = {
+      sortedInfo: {
+        columnKey: "created",
+        order: "descend",
+      },
+      prevSearchQuery: "",
+    };
+  }
+
+  handleChange = (pagination: Object, filters: Object, sorter: Object) => {
+    this.setState({
+      sortedInfo: sorter,
+    });
+  };
+
   render() {
+    const filteredDataSource = Utils.filterWithSearchQueryOR(
+      this.props.datasets,
+      ["name", "description"],
+      this.props.searchQuery,
+    );
+
+    const { sortedInfo } = this.state;
+    const sortedDataSource =
+      // Sort using the dice coefficient if the table is not sorted otherwise
+      // and if the query is longer then 3 characters to avoid sorting *all* datasets
+      this.props.searchQuery.length > 3 && sortedInfo.columnKey == null
+        ? _.chain(filteredDataSource)
+            .map(row => ({
+              row,
+              diceCoefficient: dice(row.name, this.props.searchQuery),
```

Comment:
Searching for some word of the description doesn't work for me anyways, neither in the advanced nor on the spotlight dataset view, not sure why.

Reply:
> Searching for some word of the description doesn't work for me anyways, neither in the advanced nor on the spotlight dataset view, not sure why.

Really? It works for me :thinking: 

> I'm not sure whether this will result in strange results if the user searches for a word in the description. The search results are filtered using the name and description properties, but the diceCoefficient only takes into account the dataset name for sorting.

My assumption is that the search is only used for dataset name matching, anyway. In the rare case, that a user searches the description, the ordering might be suboptimal (or "random"?), but I don't think that's a showstopper, as the user's had to scroll through a long list before anyway. We can remove the description from the search parameters, but I'd leave it as is for now until someone complains. That way, we don't remove existing functionality from the view.

Reply:
I assumed that the string `Original data and segmentation` is the description and I could search for parts of that, but that string is only a default string that is shown if there is no description. Setting a custom description and searching for it works as intended, all fine :)
Let's leave the description in there, your argumentation makes sense to me!